### PR TITLE
bump to python 3.11

### PR DESCRIPTION
**Pull request by adeebshihadeh**

### CODE REVIEW

File: laika/__init__.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import os
 
 from .astro_dog import AstroDog
-assert AstroDog
+assert AstroDog # type: ignore
```
Comment (adeebshihadeh) [inline laika/__init__.py:5]: why is this needed now?

Comment (maxime-desroches) [reply laika/__init__.py:5]: mypy will throw https://mypy.readthedocs.io/en/latest/error_code_list.html?highlight=truthy-function#check-that-function-isn-t-used-in-boolean-context-truthy-function since normally the name of an existing class (AstroDog) will always evaluate to True.

I didn't wanted to remove the test

Comment (adeebshihadeh) [reply laika/__init__.py:5]: we need to be very careful with ignores. this one seems to silently turn off a bunch of coverage in the `astro_dog.py` file
